rakelib/release.rake

```diff
@@ -0,0 +1,158 @@
+# encoding: utf-8
+require "fileutils"
+
+LOGSTASH_BUILD_PATH = File.expand_path(File.join(File.dirname(__FILE__), "..", "build"))
+
+LOGSTASH_CORE_PLUGIN_API_DEFAULT_VERSION = "1.0"
+LOGSTASH_CORE_PLUGIN_API_VERSION_FILE = File.expand_path(File.join(File.dirname(__FILE__), "..", "logstash-core-plugin-api", "lib", "logstash-core-plugin-api", "version.rb"))
+LOGSTASH_CORE_PLUGIN_API_GEMSPEC = File.expand_path(File.join(File.dirname(__FILE__), "..", "logstash-core-plugin-api", "logstash-core-plugin-api.gemspec"))
+
+GEM_BINARY_PATH = File.expand_path(File.join(File.dirname(__FILE__), "..", "vendor", "jruby", "bin", "gem"))
+
+namespace :release do
+  namespace "version" do
+    task "bump-logstash-core-plugin-api" => ["dependency:gems"] do
+      require "gems"
+      update_logstash_core_api
+    end
+  end
+
+  namespace "build" do
+    task "gems" do
+      Rake::Task["release:build:logstash-core-plugin-api"].invoke
+    end
+
+    task "logstash-core-plugin-api" do
+      Rake::Task["release:version:bump-logstash-core-plugin-api"].invoke
+      build_gemspec(LOGSTASH_CORE_PLUGIN_API_GEMSPEC)
+    end
+  end
+
+  task "publish-gems" do
+    clear_gems
+    Rake::Task["release:build:gems"].invoke
+    publish_gems
+  end
+end
+
+###
+def publish_gems(target = LOGSTASH_BUILD_PATH)
+  Dir.glob(File.join(LOGSTASH_BUILD_PATH, "*.gem")).each do |gem|
+    # If the credentials file exist use it
+    if File.exist?(File.expand_path("~/.gem/credentials"))
+      cmd = "#{GEM_BINARY_PATH} push #{gem}"
+      puts "Running: #{cmd}"
+    elsif !ENV["RUBYGEMS_API_KEY"].nil?
+      cmd = "#{GEM_BINARY_PATH} push #{gem}"
+    else
+      puts "Cannot publish the gem without a `~/.gem/credentials` or a `$RUBYGEM_API_KEY` defined"
+      exit
+    end
+
+    puts "Publishing gem: #{File.basename(gem)}"
+    #system(cmd)
```

Comment:
I have explicitly disabled this for reviewing this PR. I dont want to create a rogue gems on rubygems.